aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -415,13 +416,18 @@ impl BlockAptosVM {
         BLOCK_EXECUTOR_CONCURRENCY.set(config.local.concurrency_level as i64);
         let executor = BlockExecutor::<
             SignatureVerifiedTransaction,
-            AptosExecutorTask<S>,
+            AptosExecutorTask,
             S,
             L,
             ExecutableTestType,
         >::new(config, executor_thread_pool, transaction_commit_listener);
 
-        let ret = executor.execute_block(state_view, signature_verified_block, state_view);
+        let environment = Environment::new(state_view).try_enable_delayed_field_optimization();
```

Comment:
so we've replaced condition of (flag being enabled) && (view being capable to handle delayed fields), with us going through this path to execute block.

Is that equivalent? seems nicer either way

Reply:
I think so! We pass true in executor (equivalent to even calling the function), and in VM constructor we were checking if the feature is on, and now we just do it here.